parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -187,11 +192,31 @@ private void initializeInternalReader(ParquetInputSplit split, Configuration con
             + " in range " + split.getStart() + ", " + split.getEnd());
       }
     }
+
+    checkDeltaByteArrayProblem(footer.getFileMetaData(), configuration, filteredBlocks.get(0));
```

Comment:
Hi @rdblue I'm trying to upgrade scalding to parquet 1.8.0 and I'm seeing an `java.lang.IndexOutOfBoundsException: Index: 0, Size: 0` here in a test that runs on a local mini-cluster. 

Is it a valid assumption that this list always has a 0th element? In the case of task-side filtering, it is possible for a task to discover that there are 0 matching row groups right? I think this is a bug, but I want to make sure I'm not just missing something.